ocp-metadata/ocp-metadata.go

```diff
@@ -0,0 +1,254 @@
+	}
+	return ingressDomain, err
+}
+
+// getPrometheusURL Returns a valid prometheus endpoint from the openshift-monitoring/prometheus-k8s route
+func getPrometheusURL(dynamicClient dynamic.Interface) (string, error) {
+	route, err := dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    routeGroup,
+		Version:  routeVersion,
+		Resource: routeResource,
+	}).Namespace(monitoringNs).Get(context.TODO(), "prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	prometheusHost, found, err := unstructured.NestedString(route.UnstructuredContent(), "spec", "host")
+	if !found {
+		return "", fmt.Errorf("host field not found in %s/prometheus-k8s route spec", monitoringNs)
+	}
+	if err != nil {
+		return "", err
+	}
+	return "https://" + prometheusHost, nil
+}
+
+// getBearerToken returns a valid bearer token from the openshift-monitoring/prometheus-k8s service account
+func getBearerToken(clientset *kubernetes.Clientset) (string, error) {
+	request := authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: pointer.Int64(int64(tokenExpiration.Seconds())),
+		},
+	}
+	response, err := clientset.CoreV1().ServiceAccounts(monitoringNs).CreateToken(context.TODO(), "prometheus-k8s", &request, metav1.CreateOptions{})
+	return response.Status.Token, err
+}
+
+// getInfraDetails returns cluster anme and platform
+func (meta *Metadata) getInfraDetails() (infraObj, error) {
+	var infraJSON infraObj
+	infra, err := meta.dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "infrastructures",
+	}).Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return infraJSON, err
+	}
+	infraData, _ := infra.MarshalJSON()
+	err = json.Unmarshal(infraData, &infraJSON)
+	return infraJSON, err
+}
+
+// getVersionInfo obtains OCP and k8s version information
+func (meta *Metadata) getVersionInfo() (versionObj, error) {
+	var cv clusterVersion
+	var versionInfo versionObj
+	version, err := meta.clientSet.ServerVersion()
+	versionInfo.k8sVersion = version.GitVersion
+	if err != nil {
+		return versionInfo, err
+	}
+	clusterVersion, err := meta.dynamicClient.Resource(
+		schema.GroupVersionResource{
+			Group:    "config.openshift.io",
+			Version:  "v1",
+			Resource: "clusterversions",
+		}).Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		return versionInfo, err
+	}
+	clusterVersionBytes, _ := clusterVersion.MarshalJSON()
+	json.Unmarshal(clusterVersionBytes, &cv)
+	for _, update := range cv.Status.History {
+		if update.State == completedUpdate {
+			// obtain the version from the last completed update
+			versionInfo.ocpVersion = update.Version
+			break
+		}
+	}
+	return versionInfo, err
+}
+
+// getNodesInfo returns node information
+func (meta *Metadata) getNodesInfo() (nodeInfo, error) {
```

Comment:
Should we add the region each node is in?

Reply:
We seem to capture this with the infraObj.

Reply:
Done, at the moment it only captures the AWS region, seems like the infrastructure object from Azure or GCP does not provide that info. We'll have to implement different alternatives in the future (getting it from the node labels, machine set etc...)